inventory_management_system_api/cli/create.py

```diff
@@ -0,0 +1,38 @@
+"""Module for providing a subcommand for creating entities in IMS."""
+
+import re
+import sys
+
+import typer
+from rich.prompt import Prompt
+
+from inventory_management_system_api.cli.core import console, display_indexed_system_types
+from inventory_management_system_api.core.database import get_database
+from inventory_management_system_api.repositories.system_type import SystemTypeRepo
+from inventory_management_system_api.services.system_type import SystemTypeService
+
+app = typer.Typer()
+
+
+@app.command()
+def system_type():
+    """Creates a system type."""
+
+    # Acquire the required services/collections
+    database = get_database()
+    system_type_service = SystemTypeService(SystemTypeRepo(database))
+    system_types_collection = database.system_types
+
+    # Display a table of existing system types for reference
+    current_system_types = system_type_service.list()
+    console.print("Below is the current list of system types available:")
+    display_indexed_system_types(current_system_types)
+
+    # Obtain name of the new system type and ensure it doesn't already exist (case insensitive)
+    new_system_type_value = Prompt.ask("Please enter the value of the new system type")
+    if system_types_collection.find_one({"value": re.compile(new_system_type_value, re.IGNORECASE)}):
+        sys.exit("Already exists!")
```

Comment:
I would change this to:
```suggestion
        sys.exit("A system type with the same value already exists!")
```
Maybe also make the text colour red if that is possible?

Reply:
Here I definitely can, unfortunately typer/rich don't have an exit that will take formatting. So I didn't bother initially due to the migrations which currently have some unit tests relying on the messages from the SystemExit exception raised to know what happened. To make this work I would need to do
```python
console.print("[red]A system type with the same value already exists![/]")
typer.Exit(1)
```
I will see if I can come up with something that will work for both.

Reply:
I have currently created a generic `MigrationError` to use in the migrations code, then caught it in the CLI to handle the printing/exit so all exit messages are now printed in red.